go/vt/wrangler/materializer.go

```diff
@@ -290,6 +257,42 @@ func (wr *Wrangler) MoveTables(ctx context.Context, workflow, sourceKeyspace, ta
 		return err
 	}
 
+	// Now that the streams have been successfully created, let's put the associated
```

Comment:
Should we be moving this closer to the end just before MoveTables starts the streams? There are still validations like source time zone and checking for existing journals, that are done after this point, which can error out. Or maybe we should be moving those validations before we create the streams, since otherwise we will end up with orphan streams.

Reply:
By this point the streams/_vt.vreplication records have been created as it happens through `prepareMaterializerStreams()` here: https://github.com/vitessio/vitess/blob/ee7b9e7fbaef309d5cb39991252d88418092ccd9/go/vt/wrangler/materializer.go#L956-L980

But as you can see, for partial migrations we create the default shard routing rules right there. So it probably does make sense to do those in the same location, and in both cases *after* createStreams finishes w/o error. So I think that the current location for the routing rules is OK, but we should also move the default shard routing rules creation to the same place. What do you think?

Reply:
Ah, hadn't see that the shard routing rules were done while preparing streams. You are right: these should be in one place. 

I was asking if we should move 
```
if sourceTimeZone != "" {
		if err := mz.checkTZConversion(ctx, sourceTimeZone); err != nil {
			return err
		}
	}

	tabletShards, err := wr.collectTargetStreams(ctx, mz)
	if err != nil {
		return err
	}

	migrationID, err := getMigrationID(targetKeyspace, tabletShards)
	if err != nil {
		return err
	}

	if externalCluster == "" {
		exists, tablets, err := wr.checkIfPreviousJournalExists(ctx, mz, migrationID)
		if err != nil {
			return err
		}
		
	}
```
above the call to prepare streams.

Reply:
@rohit-nayak-ps what do you think about adding a defer which does a `MoveTables Cancel` if the error returned is not nil?

Reply:
https://github.com/vitessio/vitess/pull/13737/commits/d7360b819b20de35edeb217a82b8a2b49dafba65